source/base/parameter_handler.cc

```diff
@@ -3101,16 +3080,45 @@ MultipleParameterLoop::~MultipleParameterLoop ()
 
 
 
-bool MultipleParameterLoop::read_input (std::istream &input,
-                                        const std::string &filename,
-                                        const std::string &last_line)
+bool
+MultipleParameterLoop::read_input (std::istream &input,
+                                   const std::string &filename,
+                                   const std::string &last_line)
+{
+
+  try
+    {
+      MultipleParameterLoop::parse_input(input, filename, last_line);
+      return true;
+    }
+  catch (const ExcIO &exc)
+    {
+      throw;
+    }
+  // This catch block more or less duplicates the old behavior of this function,
+  // which was to print something to stderr for every parsing error (which are
+  // now exceptions) and then return false.
+  catch (const ExceptionBase &exc)
+    {
+      std::cerr << exc.what() << std::endl;
+      return false;
+    }
+}
+
+
+
+void
+MultipleParameterLoop::parse_input (std::istream &input,
+                                    const std::string &filename,
+                                    const std::string &last_line)
 {
   AssertThrow (input, ExcIO());
 
-  bool x = ParameterHandler::read_input (input, filename, last_line);
-  if (x)
-    init_branches ();
-  return x;
+  // Note that (to avoid infinite recursion) we have to explicitly call the
```

Comment:
can you change this to something like "call the base class version of parse_input" ?